arroyo-worker/src/lib.rs

```diff
@@ -316,6 +332,125 @@ impl WorkerServer {
     pub async fn start(self) -> Result<(), Box<dyn std::error::Error>> {
         self.start_async().await
     }
+
+    async fn spawn_control_thread(
+        &self,
+        mut shutdown_rx: tokio::sync::broadcast::Receiver<bool>,
+        mut control_rx: Receiver<ControlResp>,
+        worker_id: WorkerId,
+        job_id: String,
+    ) -> Result<()> {
+        let mut controller = ControllerGrpcClient::connect(self.controller_addr.clone()).await?;
+        tokio::spawn(async move {
+            let mut tick = tokio::time::interval(Duration::from_secs(5));
+            tick.set_missed_tick_behavior(tokio::time::MissedTickBehavior::Skip);
+            loop {
+                select! {
+                        msg = control_rx.recv() => {
```

Comment:
nit: indentation within the select is off (rustfmt ignores everything inside macros)